### PR TITLE
Support multiple inputs

### PR DESCRIPTION
**Pull request by ruizhilijhu**
modified data2json.sh and io_utils.py to support multiple parallel input cases.

### CODE REVIEW

**Comment by ruizhilijhu** [comment]
Does failing in codecov/patch prevent the PR from merging

**Comment by sw005320** [comment]
`codecov` aims to increase unit test coverage. It would be good to increase the coverage but this is not mandatory.

**Comment by ruizhilijhu** [comment]
Thanks! I will do a PR after this to support multi-encoder.

**Comment by sw005320** [comment]
I think this is related to multichannel ASR. @ShigekiKarita and @kamo-naoyuki, do you have any comments for this extension?

**Comment by ShigekiKarita** [comment]
Thanks for your fine-grained PR! I will check the detail later today. But can you add tests? Even stupid cases really help us understand what these changes are for.

For example, the `LoadInputsAndTargets` is tested at https://github.com/espnet/espnet/blob/master/test/test_utils.py#L58

Sadly, data2json does not have any existing test yet but you can refer this
https://github.com/espnet/espnet/blob/master/test_utils/test_scp2json_py.bats
how to write test for scripts.

**Comment by ruizhilijhu** [comment]
@ShigekiKarita Thanks for the suggestions. I am trying to write test for data2json.sh, but there's a lot of KALDI dependencies inside data2json.sh. For example, a lot of files used in utils/ which is a softlink to kaldi. Any suggestions to this?

**Comment by ShigekiKarita** [comment]
Thanks for adding the tests. It looks really helfull!

> For example, a lot of files used in utils/ which is a softlink to kaldi. Any suggestions to this?

Hmm. You can see CircleCI failed while Travis succeeded because Travis executes `ci/install_kaldi.sh` before `ci/test_shell.sh`. On the other hand, circle ci does not. So, can you put these lines in `ci/test_shell.sh`?
```bash
#!/usr/bin/env bash

if [ ! -e tools/kaldi ]; then
    git clone https://github.com/kaldi-asr/kaldi --depth 1 tools/kaldi
fi

...
```

If this also fails, I suggest that:
- move `ci/install_kaldi.sh` before `ci/test_shell.sh` in circleci that installs our prebuilt kaldi
- remove `ci/test_shell.sh` from centos7 image in circleci because we did not provide prebuilt kaldi for CentOS7 image (so it cannot execute `ci/install_kaldi.sh`).


I think I am the main CentOS7 user in the ESPnet but testing data2json.sh has higher priority than testing shell scripts in CentOS7. (@kan-bayashi maybe you are also CentOS7 user? Do you agree?)

**Comment by ruizhilijhu** [comment]
@ShigekiKarita Thanks for all the help! It seems like it works out finally!

File: espnet/utils/io_utils.py [hunk 1]
```diff
@@ -164,10 +164,10 @@ def __call__(self, batch):
             raise NotImplementedError
 
         if self.preprocessing is not None:
-            # Apply pre-processing only to input1 feature, now
-            if 'input1' in return_batch:
-                return_batch['input1'] = \
-                    self.preprocessing(return_batch['input1'], uttid_list,
+            # Apply pre-processing all input features
+            for x_name in [i for i in return_batch.keys() if i.startswith("input")]:
```
Comment (ShigekiKarita) [inline espnet/utils/io_utils.py]:
```python
for x_name in return_batch.keys():
    if x_name.startswith("input"):
        return_batch[x_name] = self.preprocessing(
            return_batch[x_name], uttid_list, **self.preprocess_args)
```
I think this more readable.

**Comment by ShigekiKarita** [comment]
Looks great except for one tiny comment.

**Comment by sw005320** [comment]
Thanks, @ruizhilijhu!

**Pull request merged**